test/k8sT/Services.go

```diff
@@ -146,4 +147,162 @@ var _ = Describe("K8sServicesTest", func() {
+				}
+			}
+			return baseString
+		}
+
+		// formatAPI is a helper function which formats a URI to access.
+		formatAPI := func(service, port, resource string) string {
+			if resource != "" {
+				return fmt.Sprintf("%s:%s/%s", service, port, resource)
+			}
+
+			return fmt.Sprintf("%s:%s", service, port)
+		}
+
+		ciliumPodK8s1, err := kubectl.GetCiliumPodOnNode(helpers.KubeSystemNamespace, helpers.K8s1)
+		Expect(err).Should(BeNil())
+
+		ciliumPodK8s2, err := kubectl.GetCiliumPodOnNode(helpers.KubeSystemNamespace, helpers.K8s2)
+		Expect(err).Should(BeNil())
+
+		bookInfoPath := kubectl.ManifestGet(bookinfo)
+		res := kubectl.Create(bookInfoPath)
+		defer kubectl.Delete(bookInfoPath)
+		res.ExpectSuccess()
+
+		pods, err := kubectl.WaitforPods(helpers.DefaultNamespace, formatLabelArgument(version, v1), helpers.HelperTimeout)
+		Expect(pods).Should(BeTrue())
+		Expect(err).Should(BeNil())
+
+		pods, err = kubectl.WaitforPods(helpers.DefaultNamespace, formatLabelArgument(version, v2), helpers.HelperTimeout)
+		Expect(pods).Should(BeTrue())
+		Expect(err).Should(BeNil())
+
+		reviewsPodV1, err := kubectl.GetPods(helpers.DefaultNamespace, formatLabelArgument(app, reviews, version, v1)).Filter(podNameFilter)
+		Expect(err).Should(BeNil())
+
+		productpagePodV1, err := kubectl.GetPods(helpers.DefaultNamespace, formatLabelArgument(app, productPage, version, v1)).Filter(podNameFilter)
+		Expect(err).Should(BeNil())
+
+		areEndpointsReady := kubectl.CiliumEndpointWait(ciliumPodK8s1)
+		Expect(areEndpointsReady).Should(BeTrue())
+
+		pods, err = kubectl.WaitForServiceEndpoints(helpers.DefaultNamespace, "", details, apiPort, helpers.HelperTimeout)
+		Expect(pods).Should(BeTrue())
+		Expect(err).Should(BeNil())
+
+		pods, err = kubectl.WaitForServiceEndpoints(helpers.DefaultNamespace, "", ratings, apiPort, helpers.HelperTimeout)
+		Expect(pods).Should(BeTrue())
+		Expect(err).Should(BeNil())
+
+		pods, err = kubectl.WaitForServiceEndpoints(helpers.DefaultNamespace, "", reviews, apiPort, helpers.HelperTimeout)
+		Expect(pods).Should(BeTrue())
+		Expect(err).Should(BeNil())
+
+		pods, err = kubectl.WaitForServiceEndpoints(helpers.DefaultNamespace, "", productPage, apiPort, helpers.HelperTimeout)
+		Expect(pods).Should(BeTrue())
+		Expect(err).Should(BeNil())
+
+		By("Before policy import; all pods should be able to connect")
+		shouldConnect(reviewsPodV1.String(), formatAPI(ratings, apiPort, health))
+		shouldConnect(reviewsPodV1.String(), formatAPI(ratings, apiPort, ""))
+
+		shouldConnect(productpagePodV1.String(), formatAPI(details, apiPort, health))
+		shouldConnect(productpagePodV1.String(), formatAPI(details, apiPort, ""))
+		shouldConnect(productpagePodV1.String(), formatAPI(ratings, apiPort, health))
+		shouldConnect(productpagePodV1.String(), formatAPI(ratings, apiPort, ""))
+
+		var policyPath string
+		var policyCmd string
+
+		if helpers.GetCurrentK8SEnv() != "1.6" {
```

Comment:
This Policy should have the same name (CNP.yaml) and should be one general, and the version 1.6 into the 1.6/directory. 

kubectl.ManifestGet take care about the version.

I think that one of the goals is to eliminate the If clauses on test ;-)

Reply:
Fixed.